### PR TITLE
[REPL] Add deprecation info to the REPL completion

### PR DESCRIPTION
**Pull request by ileasile**

### CODE REVIEW

File: plugins/scripting/scripting-ide-services/src/org/jetbrains/kotlin/scripting/ide_services/compiler/impl/KJvmReplCompleter.kt [hunk 1]
```diff
@@ -238,22 +241,34 @@ private class KJvmReplCompleter(
...
                                 presentableText
                             )
+                        val deprecationLevel = descriptor.annotations
+                            .findAnnotation(FqName("kotlin.Deprecated"))
+                            ?.let {
+                                val valuePair = it.argumentValue("level")?.value as? Pair<*, *>
+                                val valueName = (valuePair?.second as? Name)?.identifier ?: return@let DeprecationLevel.WARNING
+                                try {
+                                    DeprecationLevel.valueOf(valueName)
+                                } catch (e: IllegalArgumentException) {
```
Comment (ligee) [inline plugins/scripting/scripting-ide-services/src/org/jetbrains/kotlin/scripting/ide_services/compiler/impl/KJvmReplCompleter.kt:259]: Please add a comment, why such a catch is needed here. It looks hacky to me.

**Review by ligee — APPROVED** [review]
I would suggest making the commit message a bit clearer, e.g. "Add Kotlin `DeprecationLevel` info the REPL completion"

File: plugins/scripting/scripting-ide-services/src/org/jetbrains/kotlin/scripting/ide_services/compiler/impl/KJvmReplCompleter.kt (hunk 1 above)
Comment (ileasile) [reply plugins/scripting/scripting-ide-services/src/org/jetbrains/kotlin/scripting/ide_services/compiler/impl/KJvmReplCompleter.kt:259]: Changed the logic here a bit, removed try/catch

**Comment by ileasile** [comment]
Merged manually: e9af2849efda45ef462be4f45f14e78fbeaca2e6

**Pull request closed without merging**